tracer/src/Datadog.Trace/AppSec/BlockException.cs

```diff
@@ -30,17 +30,20 @@ internal BlockException(System.Runtime.Serialization.SerializationInfo info, Sys
         {
         }
 
-        public BlockException(IResult result, bool reported = false)
+        public BlockException(string triggerData, ulong aggregatedTotalRuntime, ulong aggregatedTotalRuntimeWithBindings, bool reported = false)
```

Comment:
Since https://github.com/DataDog/dd-trace-dotnet/pull/3608 , not sure it's necessary breaking down into primitives but it might still be a good solution to avoid reading `Data` if not necessary.  In this case, it might be good to make `IResult` disposable again as this could feel like a bit of a [hack ](https://github.com/DataDog/dd-trace-dotnet/blob/f3ddb26467987c70f24b796affd4a896ef800281/tracer/src/Datadog.Trace/AppSec/Waf/Result.cs#L30) , (was to only marshall if it were to be read, so reported...)

Reply:
Agree this isn't necessary any more, but now the change is in place, there didn't seem to be a huge amount value in rolling it back. I can do, if you feel strongly it's better to store the result object in the exception.